crates/bin/prove_block/src/main.rs

```diff
@@ -124,21 +124,26 @@ async fn main() -> Result<(), Box<dyn Error>> {
             panic!("Block is still pending!");
         }
     };
-    let older_block =
-        match provider.get_block_with_txs(BlockId::Number(block_number - STORED_BLOCK_HASH_BUFFER)).await.unwrap() {
-            MaybePendingBlockWithTxs::Block(block_with_txs) => block_with_txs,
-            MaybePendingBlockWithTxs::PendingBlock(_) => {
-                panic!("Block is still pending!");
-            }
-        };
+
+    // We only need to get the older block number and hash. No need to fetch all the txs
+    let older_block = match provider
+        .get_block_with_tx_hashes(BlockId::Number(block_number - STORED_BLOCK_HASH_BUFFER))
+        .await
+        .unwrap()
```

Comment:
For another PR: now that we use `anyhow`, these unwraps should be replaced by `.with_context(|| format!("failed to retrieve previous block ({})", previous_block_id))`